server/lib/notifications.js

```diff
@@ -201,14 +201,48 @@ async function notifyByEmail(activity) {
 
     case activityType.COLLECTIVE_UPDATE_PUBLISHED:
       twitter.tweetActivity(activity);
-      activity.data.update = await models.Update.findByPk(activity.data.update.id, {
-        include: [{ model: models.Collective, as: 'fromCollective' }],
-      });
-      activity.data.update = activity.data.update.info;
-      notifyMembersOfCollective(activity.data.update.CollectiveId, activity, {
-        from: `${activity.data.collective.name}
-        <no-reply@${activity.data.collective.slug}.opencollective.com>`,
-      });
+
+      activity.data.collective = await models.Collective.findByPk(activity.data.collective.id);
+      activity.data.hostedCollectives = activity.data.collective.getHostedCollectives();
+
+      if (activity.data.collective.info.isHostAccount) {
+        switch (activity.data.update.updateNotificationAudience) {
+          case 'Hosted Collective Admins':
+            activity.data.hostedCollectives.map(collective => {
```

Comment:
The Open Source Collective is hosting 1200+ collectives. `notifyAdminsOfCollective` triggers at least 2 SQL queries (and usually more) so this loop will trigger more than 2500 SQL queries at the same time, which is really too much stress for the DB, we need to optimize this. Ideally we should have one single SQL query that loads all the hosted collective admins emails at once. A new `notifyHostedCollectiveAdmins` helper would be a good way to achieve that.